### ISSUE TITLE
Add tests that ensure volatile semantics are actually applied across threads

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Core#342]

**Comment by minborg** [comment]
What is the contract for the volatile operations (e.g. if applied across primitive size alignments and cache-line sizes)?

````
[Reader 0@61] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f286bd (61) was efffffff is now f0000000
[Reader 3@61] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f286bd (61) was efffffff is now f0000000
[Reader 4@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was efffffff is now f0000000
[Reader 3@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was efffffff is now f0000000
[Reader 1@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was f0000000 is now f0000001
[Reader 0@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was f0000000 is now f0000001
[Reader 2@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was f0000000 is now f0000001
[Reader 4@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was f0000001 is now f7ffffff
[Reader 1@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was f0000001 is now f7ffffff
[Reader 0@63] WARN net.openhft.chronicle.core.UnsafeMemory - Int@600003f287bf (63) was ffffff is now 1000000
````

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.23ea13](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.23ea13), [BOM-2.23ea69](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23ea69)
